pkg/apis/wsecurity/v1/v1_test.go

```diff
@@ -0,0 +1,317 @@
+package v1
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+)
+
+// Utility function to convert []interface{} to [][]interface{}
+func toSlice(slices []interface{}) [][]interface{} {
+	result := make([][]interface{}, len(slices))
+	for sliceIndex := 0; sliceIndex < len(slices); sliceIndex++ {
+		s := reflect.ValueOf(slices[sliceIndex])
+		if s.Kind() != reflect.Slice {
+			panic("InterfaceSlice() given a non-slice type")
+		}
+
+		// Keep the distinction between nil and empty slice input
+		if s.IsNil() {
+			return nil
+		}
+
+		converted := make([]interface{}, s.Len())
+
+		for i := 0; i < s.Len(); i++ {
+			converted[i] = s.Index(i).Interface()
+		}
+		result[sliceIndex] = converted
+	}
+	return result
+}
+
+func ValueTests_Test(t *testing.T, profiles []ValueProfile, piles []ValuePile, configs []ValueConfig, arguments ...interface{}) {
+	args := toSlice(arguments)
+
+	t.Run("Basics", func(t *testing.T) {
+
+		// Initial Tests
+		profiles[9].String(9)
+		piles[9].String(9)
+		configs[9].String(9)
+		piles[9].Clear()
+		piles[3].Merge(piles[4])
+		configs[3].Fuse(configs[4])
+		configs[5].Learn(piles[5])
+		configs[6].Decide(profiles[0])
+		piles[6].Clear()
+		piles[7].String(7)
+		configs[7].String(7)
+		profiles[2].String(7)
```

Comment:
I don't understand how these permutations were arrived at. Also, why call `.String(9)` without testing the output? I'd expect `String` to not change the value; if these are simply testing for non-panic, it seems like a more focused test would make more sense.

Reply:
We need more focused tests. 

Maybe it is best to leave this as is for now and address it in a separate PR. I think there is more to be done in improving the current tests.

(We do have many focused tests already, these were added to capture some additional code branches like when you Sting() a freshly created object )